internal/driver/glfw/canvas.go

```diff
@@ -386,6 +384,20 @@ func (c *glCanvas) objectTrees() []fyne.CanvasObject {
 	return trees
 }
 
+func (c *glCanvas) overlayChanged(focusMgr, prevFocusMgr *app.FocusManager) {
+	c.Lock()
+	defer c.Unlock()
```

Comment:
nit: It might be better to not use a defer here, and instead move `c.Unlock()` down to after line 396. In case dev is doing something weird in Focus{Lost|Gained} which could cause deadlock